### DISCUSSION TITLE
Upgrade from zeus-l4t-r32.3.1 to dunfell-l4t-r32.4.3: Things to check?

### DISCUSSION THREAD

**Discussion by manuel-wagesreither** [OE4T/meta-tegra#524] [General]
Hello everyone!

We're going from [zeus-l4t-r32.3.1](https://github.com/OE4T/meta-tegra/tree/zeus-l4t-r32.3.1) to [dunfell-l4t-r32.4.3](https://github.com/OE4T/meta-tegra/tree/dunfell-l4t-r32.4.3) and I want to make sure that we do not run into unexpected problems.

@dwalkes created [this post on the nvidia forum regarding the location of the U-Boot environment](https://forums.developer.nvidia.com/t/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/82113/13), but I'm of the understanding that the issue Dan was worried about is resolved. [NVIDIA provided a patch](https://forums.developer.nvidia.com/t/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/82113/14), and we already have a [patch doing the same](https://forums.developer.nvidia.com/t/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/82113/15). Hence, we're fine here.

Are there other things one needs to be concerned about?

I compared the `tegraflash.zip/flash.xml` (zeus) with `tegraflash.tar.gz/flash.xml.in` (there was no `flash.xml`) (dunfell). There are a few differences, but they are small enough for me not to bother.

The differences are:
- In `tegraflash.tar.gz/flash.xml.in` (dunfell) most variables are replaced by their values in-place. This is in contrast to zeus, were there were separate files. At dunfell, there is no `tegraflash.tar.gz/flash.xml`.
- Exceptions are `DATAFILE`, `APPFILE`, `BPFDTB-FILE` and `VERFILE`. They show up un-replaced in dunfells `tegraflash.tar.gz/flash.xml.in`.

Still, I was able to flash this dunfell image just fine. So it seems this is only of cosmetic nature.

Is there anything else I should have an eye on? I'm especially interested on things breaking OTA updateability.

Thanks for you support!
Best regards,
Manuel

**Comment by manuel-wagesreither** [comment]
If someone finds this through some search engine: When you upgrade to dunfell and use mender as OTA update solution, take care to use poky of version >= `dunfell-23.0.3` bringing `libubootenv 0.3`. Usage of older poky version will [make you run into this problem](https://github.com/OE4T/meta-mender-community/issues/9).

[If you're using tegra-demo-distro, you are fine, though](https://github.com/OE4T/meta-mender-community/issues/9#issuecomment-739542110). It is not affected.

**Comment by dwalkes** [comment]
@manuel-wagesreither you are on `jetson-tx2` with u-boot, correct?  If so the only thing I know about to look for is u-boot environment location, which can't change between releases without some additional work like the work @madisongh did at https://github.com/Trellis-Logic/meta-mender-community-old/pull/10/commits/7536c741444b4add6f48a84c97bf815b93142170 and, based on the thread with NVIDIA you referenced, shouldn't be necessary to relocate on `jetson-tx2`.

If I were going to deploy this I'd want to make sure that I verified the zeus -> dunfell transition starting in either boot slot.  So I'd have a test case which tegraflashed at zeus and verified the upgrade worked, then one which tegraflashed at zeus, upgraded at zeus to a new boot slot while still on the zeus branch, then upgraded to dunfell.

The second thing I'd check is rollback, I'd want to make sure I could reboot > 10 times after upgrading across zeus->dunfell and that nothing in the NVIDIA bootchain was attempting to roll back to the old boot slot.  I've got a start on some scripts which can help with this for cboot cases at https://github.com/mendersoftware/meta-mender-community/pull/207 and I'd like to expand these to cover more cases and especially u-boot after some additional u-boot changes like the one at https://github.com/OE4T/meta-mender-community/pull/8#issuecomment-744376127 are in.

**Reply by manuel-wagesreither** [reply]
`jetson-tx2` with U-Boot, correct. Thanks a lot, I will do as you suggested.
